craft_application/services/lifecycle.py

```diff
@@ -155,10 +156,17 @@ def _init_lifecycle_manager(self) -> LifecycleManager:
         """
         emit.debug(f"Initialising lifecycle manager in {self._work_dir}")
         emit.trace(f"Lifecycle: {repr(self)}")
-        feature_args: dict[str, Any] = {}
 
-        if self._app.features.package_repository:
-            feature_args["package_repositories"] = self._project.package_repositories
+        if self._app.features.package_repository and issubclass(
```

Comment:
I think this `issubclass()` check should happen earlier, maybe on a new `_validate_metadata` method called in Application's constructor. And it should be an error to have `self._app.features.package_repository` enabled but a non-mixin project
then you don't need to check the subclass here (though there's no harm I guess)

Reply:
I think Alex does not want put checks in the Application's constructor. Maybe I can put in the AppMetadata's `__post_init__`?

Reply:
hmm but then `__post_init__` seems like the wrong place too
maybe early in `run()`?

Reply:
OK then